### PR TITLE
feat: multiple domains

### PR DESCRIPTION
**Pull request by fgerschwiler**

### CODE REVIEW

File: internal/org/repository/view/model/org_domain.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+)
+
+const (
+	OrgDomainKeyOrgID  = "org_id"
+	OrgDomainKeyDomain = "domain"
+)
+
+type OrgDomainView struct {
+	Domain   string `json:"domain" gorm:"column:domain;primary_key"`
+	OrgID    string `json:"-" gorm:"column:org_id"`
```
Comment (livio-a) [inline internal/org/repository/view/model/org_domain.go:20]: unless they're verified domains could exist on different orgs

File: internal/org/repository/view/org_domain_view.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+import (
+	global_model "github.com/caos/zitadel/internal/model"
+	org_model "github.com/caos/zitadel/internal/org/model"
+	"github.com/caos/zitadel/internal/org/repository/view/model"
+	"github.com/caos/zitadel/internal/view"
+	"github.com/jinzhu/gorm"
+)
+
+func OrgDomainByDomain(db *gorm.DB, table, domain string) (*model.OrgDomainView, error) {
```
Comment (livio-a) [inline internal/org/repository/view/org_domain_view.go:11]: ...then orgID will be needed as param

File: internal/user/model/user.go [hunk 3]
```diff
@@ -66,10 +68,17 @@ const (
 	GENDER_DIVERSE
 )
 
-func (u *User) SetEmailAsUsername() {
-	if u.Profile != nil && u.UserName == "" && u.Email != nil {
+func (u *User) CheckOrgIamPolicy(policy *org_model.OrgIamPolicy) error {
+	if policy == nil {
+		return caos_errors.ThrowPreconditionFailed(nil, "MODEL-zSH7j", "Org Iam Policy should not be nil")
+	}
+	if policy.UserLoginMustBeDomain && strings.Contains(u.UserName, "@") {
```
Comment (livio-a) [inline internal/user/model/user.go:75]: not sure how we should check this properly

File: internal/admin/repository/eventsourcing/repository.go [hunk 4]
```diff
@@ -24,6 +24,8 @@ type Config struct {
 	Eventstore  es_int.Config
 	View        types.SQL
 	Spooler     spooler.SpoolerConfig
+	//TODO: should this be located in system-defaults?
```
Comment (livio-a) [inline internal/admin/repository/eventsourcing/repository.go:27]: .

File: internal/management/repository/eventsourcing/repository.go [hunk 5]
```diff
@@ -22,9 +22,11 @@ import (
 
 type Config struct {
 	SearchLimit uint64
-	Eventstore  es_int.Config
-	View        types.SQL
-	Spooler     spooler.SpoolerConfig
+	//TODO: should this be located in system-defaults?
```
Comment (livio-a) [inline internal/management/repository/eventsourcing/repository.go:25]: .

File: internal/org/model/org_view.go [hunk 6]
```diff
@@ -66,8 +65,8 @@ func OrgViewToOrg(o *OrgView) *Org {
 			ResourceOwner: o.ResourceOwner,
 			Sequence:      o.Sequence,
 		},
-		Domain: o.Domain,
-		Name:   o.Name,
-		State:  o.State,
+		//Domain: o.Domain,
```
Comment (livio-a) [inline internal/org/model/org_view.go:68]: please remove

File: internal/user/repository/eventsourcing/user.go [hunk 7]
```diff
@@ -631,3 +649,44 @@ func isEventValidation(aggregate *es_models.Aggregate, eventType es_models.Event
...
+				}
+			}
+		}
+		split := strings.Split(userName, "@")
+		if len(split) != 2 {
+			return nil
+		}
+		for _, d := range domains {
+			if d.Verified && d.Domain == split[1] {
+				return errors.ThrowPreconditionFailed(nil, "EVENT-us5Zw", "domain already reserverd")
```
Comment (livio-a) [inline internal/user/repository/eventsourcing/user.go:687]:
```suggestion
				return errors.ThrowPreconditionFailed(nil, "EVENT-us5Zw", "domain already reserved")
```

File: internal/admin/repository/eventsourcing/setup/setup.go [hunk 8]
```diff
@@ -193,12 +208,21 @@ func (setUp *initializer) orgs(ctx context.Context, orgs []types.Org) error {
...
+		Domains: []*org_model.OrgDomain{&org_model.OrgDomain{Domain: org.Domain}},
 	}
 	return setUp.repos.OrgEvents.CreateOrg(ctx, createOrg)
 }
 
+func (setUp *initializer) iamorgpolicy(ctx context.Context, org *org_model.Org) (*org_model.OrgIamPolicy, error) {
+	ctx = setSetUpContextData(ctx, org.AggregateID)
+	policy := &org_model.OrgIamPolicy{
+		ObjectRoot:            models.ObjectRoot{AggregateID: org.AggregateID},
+		UserLoginMustBeDomain: false,
```
[the remark this answers is not in the document]
Comment (fgerschwiler) [reply internal/admin/repository/eventsourcing/setup/setup.go:221]: default is true, this is only set if explicit set in setup (only on global org)

**Pull request merged**